src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
```

Comment:
BTW, I'm relatively sure that this StringBuffer technique doesn't give any performance benefit in modern VMs. I've been meaning to go through and remove it everywhere. So if you want to avoid using it here, I'd be in favour :-)